token/services/network/fabric/lookup/deliveryllm.go

```diff
@@ -122,23 +136,32 @@ func (p *deliveryBasedLLMProvider) NewManager(network, channel string) (Listener
 }
 
 type deliveryBasedLLM struct {
-	lm finality.ListenerManager[TxInfo]
+	lm EventsListenerManager
+}
+
+func (c *deliveryBasedLLM) PermanentLookupListenerSupported() bool {
+	return true
+}
+
+func (m *deliveryBasedLLM) AddPermanentLookupListener(namespace string, key string, listener Listener) error {
+	return m.lm.AddPermanentEventListener(key, &listenerEntry{namespace, listener})
 }
 
 func (m *deliveryBasedLLM) AddLookupListener(namespace string, key string, startingTxID string, stopOnLastTx bool, listener Listener) error {
-	return m.lm.AddFinalityListener(key, &listenerEntry{namespace, listener})
+	return m.lm.AddEventListener(key, &listenerEntry{namespace, listener})
 }
 
 func (m *deliveryBasedLLM) RemoveLookupListener(key string, listener Listener) error {
-	return m.lm.RemoveFinalityListener(key, &listenerEntry{"", listener})
+	return m.lm.RemoveEventListener(key, &listenerEntry{"", listener})
 }
 
 type endorserTxInfoMapper struct {
 	network string
-	prefix  string
+	prefix1 string
```

Comment:
Is there a guarantee that they'll never be more? We could use a slice

Reply:
moved to slice